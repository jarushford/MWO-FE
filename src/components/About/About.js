import React from 'react'
import AboutContainer from '../AboutContainer/AboutContainer'
import '../../main.scss'

export default function About() {
  return (
    <section className="about-container">
      <h1>about</h1>
      <img src="./assets/about-cover.jpg" alt="Mad Wallace in front of Denver skyline" />
      <p>Who is Mad Wallace? A person? A place? A specter in the night? You’re on the right track. Mad Wallace is a family that entered into a legally binding agreement to split everything from food and shelter, to showers and future children. Sometimes they love each other and sometimes they don’t. Mad Wallace is a couple of ornery Jakes, an unflappable Jamie, and a super nice dude named Cory. Formed in the Spring of 2017, this group has perfected the art of bringing the overwhelming energy of house parties to whatever venue they play. Mad Wallace’s dynamic sound keeps audiences dancing through rock-y, heav-y, Latin-y, and hipp-y grooves that somehow wind their way into a coherent set. Based out of Denver, Mad Wallace has made its home along the Front Range, playing at venues like Cervantes, Summit Music Hall, and Larimer Lounge while sharing the stage with bands like Perpetual Groove and Legato. However, the band is most at home in a grungy house playing into the wee hours of the nightfall.</p>
      <p className="center-p">“So if you are feeling down and kinda beat, just grab a hold of your old friend Pete.</p>
      <p className="center-p">If you are feeling hungry and need to eat, 
      just grab a hold of your old friend Pete.</p>
      <p className="center-p">If you’re feeling tired and need to sleep,
      just grab a hold of your old friend Pete.”</p>
      <p className="center-p">If you’re feeling lost and can’t find the beat, 
      Grab a hold of your old friend Pete!”</p>
      <p className="center-p"><strong>And Pete </strong>
      - Mad Wallace</p>
      <div className="photo-container">
        {
          ['Sabetta', 'Lauer', 'Jamie', 'Cory'].map(name => {
            return <AboutContainer name={name} key={name} />
          })
        }
      </div>
      <a href="http://www.ciphermagazine.com/mad-wallace/" target="_blank" rel="noopener noreferrer">
        <button>
          Want to know more about the band?
        </button>
      </a>
    </section>
  )
}