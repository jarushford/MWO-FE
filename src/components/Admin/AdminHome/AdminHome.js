import React from 'react'
import '../../../main.scss'
import { NavLink, Redirect } from 'react-router-dom'
import { connect } from 'react-redux'

export function AdminHome({ user }) {
  if (!user) {
    return <Redirect to="/admin" />
  } else {
    return (
      <div className="admin-home">
        <header>
          <NavLink to="/admin/tour" className="nav-item">Tour</NavLink>
          <NavLink to="/admin/videos" className="nav-item">Videos</NavLink>
          <NavLink to="/admin/news" className="nav-item">News</NavLink>
          <NavLink to="/admin/photos" className="nav-item">Photos</NavLink>
          <NavLink to="/admin/mailing" className="nav-item">Mailing</NavLink>
        </header>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(AdminHome)