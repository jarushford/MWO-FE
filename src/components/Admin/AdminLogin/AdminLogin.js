import React, { useState } from 'react'
import { sha256 } from 'js-sha256'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import { setUser } from '../../../actions'
import '../../../main.scss'

export function AdminLogin({ setUser }) {
  const [email, updateEmail] = useState('')
  const [password, updatePassword] = useState('')
  const [isLoggedIn, setLogin] = useState(false)
  const [hasErrored, setError] = useState(false)

  const handleSubmit = async e => {
    e.preventDefault()
    const hashedPassword = sha256(password)
    const user = { email, password: hashedPassword }
    const url = 'https://mwo-be.herokuapp.com/api/v1/login'

    const response = await fetch(url, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(user)
    })

    setUser(response.ok)
    setError(response.ok)
    setLogin(response.ok)
  }

  if (isLoggedIn) {
    return <Redirect to="/admin/tour" />
  }

  return (
    <div className="admin-login">
      <h1 className="login-title">Login</h1>
      <p className={`err-msg ${hasErrored && 'error'}`}>
        Incorrect username or password
      </p>
      <form
        className="login-form"
        data-testid="form"
        onSubmit={e => handleSubmit(e)}
      >
        <input
          type="text"
          placeholder="email"
          value={email}
          onChange={e => updateEmail(e.target.value)}
          data-testid="email"
        />
        <input
          type="password"
          placeholder="password"
          value={password}
          onChange={e => updatePassword(e.target.value)}
          data-testid="password"
        />
        <button>Login</button>
      </form>
    </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  setUser: (user) => dispatch(setUser(user))
})

export default connect(null, mapDispatchToProps)(AdminLogin)