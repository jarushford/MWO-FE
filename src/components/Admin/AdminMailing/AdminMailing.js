import React, { useState, useEffect } from 'react'
import '../../../main.scss'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'

export function AdminMailing({ user }) {
  const [contacts, setContacts] = useState([])

  const getContacts = async () => {
    const url = 'https://mwo-be.herokuapp.com/api/v1/mailing'
    const response = await fetch(url)
    if (!response.ok) {
      alert('Error fetching data, please try again later')
    } else {
      const result = await response.json()
      const contacts = result.map(contact => {
        return (
          <li key={contact.email}>{contact.email}</li>
      )})
      setContacts(contacts)
    }
  }

  useEffect(() => {
    getContacts()
  }, [contacts])

  if (!user) {
    return <Redirect to="/admin" />
  } else {
    return (
      <div className="admin-photo">
        <h1 className="admin-title">Mailing</h1>
        <ul className="mailing-container">
          {contacts}
        </ul>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(AdminMailing)