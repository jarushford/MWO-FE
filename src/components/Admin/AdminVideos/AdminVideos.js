import React, { useState, useEffect } from 'react'
import '../../../main.scss'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import VideoForm from '../../Forms/VideoForm'

export function AdminVideos({ user }) {
  const [videos, setVideos] = useState([])
  const [showForm, setForm] = useState(false)

  const getVideos = async () => {
    const url = 'https://mwo-be.herokuapp.com/api/v1/videos'
    const response = await fetch(url)
    if (!response.ok) {
      alert('Error fetching data, please try again later')
    } else {
      const result = await response.json()
      const videos = result.map(video => {
        return (
          <div key={video.id} className="video">
            <p>{video.title}</p>
            <a className="video-link" href={video.link}>{video.link}</a>
            <img src={video.thumbnail} alt="video thumbnail" />
            <button onClick={() => removeVideo(video.id)}>Remove</button>
          </div>
      )})
      setVideos(videos)
    }
  }

  const removeVideo = async (id) => {
    const url = `https://mwo-be.herokuapp.com/api/v1/videos/${id}`
    await fetch(url, {
      method: 'DELETE',
      headers: { 'Content-Type': 'application/json' }
    })
  }

  useEffect(() => {
    getVideos()
  }, [videos])

  if (!user) {
    return <Redirect to="/admin" />
  } else {
    return (
      <div className="admin-photo">
        <h1 className="admin-title">Videos</h1>
        <button
          className="add-btn"
          onClick={() => setForm(!showForm)}
        >
          Add Video
        </button>
        <VideoForm showForm={showForm} />
        <div className="video-container">
          {videos}
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(AdminVideos)