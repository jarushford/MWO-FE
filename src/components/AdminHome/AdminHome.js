import React from 'react'
import '../../main.scss'
import { NavLink, Redirect } from 'react-router-dom'
import { connect } from 'react-redux'

export function AdminHome({ user }) {
  if (!user) {
    return <Redirect to="/admin" />
  } else {
    return (
      <div className="admin-home">
        <header>
          <NavLink to="/" className="nav-item">Tour</NavLink>
          <NavLink to="/" className="nav-item">Videos</NavLink>
          <NavLink to="/" className="nav-item">News</NavLink>
          <NavLink to="/" className="nav-item">Photos</NavLink>
          <NavLink to="/" className="nav-item">Mailing</NavLink>
        </header>
        <h1 className="title">Welcome to the Mad Wallace Admin Portal</h1>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(AdminHome)