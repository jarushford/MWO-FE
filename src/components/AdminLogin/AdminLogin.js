import React, { useState } from 'react'
import { sha256 } from 'js-sha256'
import { Redirect } from 'react-router-dom'

export default function AdminLogin() {
  const [email, updateEmail] = useState('')
  const [password, updatePassword] = useState('')
  const [isLoggedIn, setLogin] = useState(false)
  const [hasErrored, setError] = useState(false)

  const handleSubmit = async e => {
    e.preventDefault()
    const hashedPassword = sha256(password)
    const user = { email, password: hashedPassword }
    const url = 'https://mwo-be.herokuapp.com/api/v1/login'
    
    const response = await fetch(url, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(user)
    })

    !response.ok
      ? setError(true)
      : setLogin(true)
  }

  if (isLoggedIn) {
    return <Redirect to="/" />
  }

  return (
    <div className="admin-login">
      <h1>Login</h1>
      <p className={`err-msg ${hasErrored && 'error'}`}>
        Incorrect username or password
      </p>
      <form
        className="login-form"
        onSubmit={e => {
          return handleSubmit(e)
        }}
      >
        <input
          type="text"
          placeholder="email"
          value={email}
          onChange={e => updateEmail(e.target.value)}
        />
        <input
          type="password"
          placeholder="password"
          value={password}
          onChange={e => updatePassword(e.target.value)}
        />
        <button>Login</button>
      </form>
    </div>
  )
}