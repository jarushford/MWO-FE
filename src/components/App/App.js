import React from 'react'
import { Route, Switch } from 'react-router-dom'
import AdminLogin from '../AdminLogin/AdminLogin'
import AdminHome from '../AdminHome/AdminHome'
import AdminTour from '../AdminTour/AdminTour'
import AdminPhotos from '../AdminPhotos/AdminPhotos'
import AdminNews from '../AdminNews/AdminNews'

export default function App() {
  return (
    <div className="app">
      <Switch>
        <Route exact path="/" />
        <Route exact path="/admin" component={AdminLogin} />
        <Route path="/admin/tour" render={() => {
          return (
            <div>
              <AdminHome />
              <AdminTour />
            </div>
          )
        }} />
        <Route path="/admin/photos" render={() => {
          return (
            <div>
              <AdminHome />
              <AdminPhotos />
            </div>
          )
        }} />
        <Route path="/admin/news" render={() => {
          return (
            <div>
              <AdminHome />
              <AdminNews />
            </div>
          )
        }} />
      </Switch>
    </div>
  )
}
