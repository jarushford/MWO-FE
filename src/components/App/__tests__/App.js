import React from 'react'
import App from '../App'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { MemoryRouter } from 'react-router-dom'
import MainNav from '../../MainNav/MainNav'
import Footer from '../../Footer/Footer'
import { Home } from '../../Home/Home'
import { Tour } from '../../Tour/Tour'
import { Media } from '../../Media/Media'
import About from '../../About/About'
import Contact from '../../Contact/Contact'
import PressKit from '../../PressKit/PressKit'
import { AdminHome } from '../../Admin/AdminHome/AdminHome'
import { AdminLogin } from '../../Admin/AdminLogin/AdminLogin'
import { AdminTour } from '../../Admin/AdminTour/AdminTour'
import { AdminPhotos } from '../../Admin/AdminPhotos/AdminPhotos'
import { AdminNews } from '../../Admin/AdminNews/AdminNews'
import { AdminVideos } from '../../Admin/AdminVideos/AdminVideos'
import { AdminMailing } from '../../Admin/AdminMailing/AdminMailing'

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correct components if on Tour route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: []
      }))}>
        <MemoryRouter initialEntries={['/tour']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(Tour)).toHaveLength(1)
    expect(wrapper.find(MainNav)).toHaveLength(1)
    expect(wrapper.find(Footer)).toHaveLength(1)
  })

  it('should render correct components if on Media route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: []
      }))}>
        <MemoryRouter initialEntries={['/media']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(Media)).toHaveLength(1)
    expect(wrapper.find(MainNav)).toHaveLength(1)
    expect(wrapper.find(Footer)).toHaveLength(1)
  })

  it('should render correct components if on About route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: []
      }))}>
        <MemoryRouter initialEntries={['/about']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(About)).toHaveLength(1)
    expect(wrapper.find(MainNav)).toHaveLength(1)
    expect(wrapper.find(Footer)).toHaveLength(1)
  })

  it('should render correct components if on Contact route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: []
      }))}>
        <MemoryRouter initialEntries={['/contact']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(Contact)).toHaveLength(1)
    expect(wrapper.find(MainNav)).toHaveLength(1)
    expect(wrapper.find(Footer)).toHaveLength(1)
  })

  it('should render correct components if on PressKit route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: []
      }))}>
        <MemoryRouter initialEntries={['/presskit']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(PressKit)).toHaveLength(1)
    expect(wrapper.find(Footer)).toHaveLength(1)
  })

  it('should render correct components if on root route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: []
      }))}>
        <MemoryRouter initialEntries={['/']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(Home)).toHaveLength(1)
    expect(wrapper.find(MainNav)).toHaveLength(1)
    expect(wrapper.find(Footer)).toHaveLength(1)
  })

  it('should render correct components if on Admin root route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: []
      }))}>
        <MemoryRouter initialEntries={['/admin']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(AdminLogin)).toHaveLength(1)
  })

  it('should render correct components if on Admin Tour route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: [],
        user: true
      }))}>
        <MemoryRouter initialEntries={['/admin/tour']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(AdminHome)).toHaveLength(1)
    expect(wrapper.find(AdminTour)).toHaveLength(1)
  })

  it('should render correct components if on Admin Tour route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: [],
        user: true
      }))}>
        <MemoryRouter initialEntries={['/admin/tour']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(AdminHome)).toHaveLength(1)
    expect(wrapper.find(AdminTour)).toHaveLength(1)
  })

  it('should render correct components if on Admin Photos route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: [],
        user: true
      }))}>
        <MemoryRouter initialEntries={['/admin/photos']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(AdminHome)).toHaveLength(1)
    expect(wrapper.find(AdminPhotos)).toHaveLength(1)
  })

  it('should render correct components if on Admin Videos route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: [],
        user: true
      }))}>
        <MemoryRouter initialEntries={['/admin/videos']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(AdminHome)).toHaveLength(1)
    expect(wrapper.find(AdminVideos)).toHaveLength(1)
  })

  it('should render correct components if on Admin News route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: [],
        user: true
      }))}>
        <MemoryRouter initialEntries={['/admin/news']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(AdminHome)).toHaveLength(1)
    expect(wrapper.find(AdminNews)).toHaveLength(1)
  })

  it('should render correct components if on Admin Mailing route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        tourDates: [],
        photos: [],
        videos: [],
        news: [],
        user: true
      }))}>
        <MemoryRouter initialEntries={['/admin/mailing']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(AdminHome)).toHaveLength(1)
    expect(wrapper.find(AdminMailing)).toHaveLength(1)
  })
})