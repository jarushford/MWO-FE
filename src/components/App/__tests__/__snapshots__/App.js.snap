// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/tour"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/media"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/about"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/contact"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/admin"
        />
        <Route
          path="/admin/tour"
          render={[Function]}
        />
        <Route
          path="/admin/photos"
          render={[Function]}
        />
        <Route
          path="/admin/news"
          render={[Function]}
        />
        <Route
          path="/admin/videos"
          render={[Function]}
        />
        <Route
          path="/admin/mailing"
          render={[Function]}
        />
        <Route
          path="/presskit"
          render={[Function]}
        />
        <Route
          component={[Function]}
          path=""
        />
      </Switch>,
      "className": "app",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/tour"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/media"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/about"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/contact"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/admin"
          />,
          <Route
            path="/admin/tour"
            render={[Function]}
          />,
          <Route
            path="/admin/photos"
            render={[Function]}
          />,
          <Route
            path="/admin/news"
            render={[Function]}
          />,
          <Route
            path="/admin/videos"
            render={[Function]}
          />,
          <Route
            path="/admin/mailing"
            render={[Function]}
          />,
          <Route
            path="/presskit"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            path=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/tour",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/media",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/about",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/contact",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/admin",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/admin/tour",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/admin/photos",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/admin/news",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/admin/videos",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/admin/mailing",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/presskit",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/tour"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/media"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/about"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/contact"
            render={[Function]}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/admin"
          />
          <Route
            path="/admin/tour"
            render={[Function]}
          />
          <Route
            path="/admin/photos"
            render={[Function]}
          />
          <Route
            path="/admin/news"
            render={[Function]}
          />
          <Route
            path="/admin/videos"
            render={[Function]}
          />
          <Route
            path="/admin/mailing"
            render={[Function]}
          />
          <Route
            path="/presskit"
            render={[Function]}
          />
          <Route
            component={[Function]}
            path=""
          />
        </Switch>,
        "className": "app",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/tour"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/media"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/about"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/contact"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/admin"
            />,
            <Route
              path="/admin/tour"
              render={[Function]}
            />,
            <Route
              path="/admin/photos"
              render={[Function]}
            />,
            <Route
              path="/admin/news"
              render={[Function]}
            />,
            <Route
              path="/admin/videos"
              render={[Function]}
            />,
            <Route
              path="/admin/mailing"
              render={[Function]}
            />,
            <Route
              path="/presskit"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              path=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/tour",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/media",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/about",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/contact",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/admin",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/admin/tour",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/admin/photos",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/admin/news",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/admin/videos",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/admin/mailing",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/presskit",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
